### PR TITLE
Convert CategoryPanel to a functional component

### PR DESCRIPTION
**Pull request by bbecquet**
## Description
The first components I converted were easy, let's continue with the trickier ones…

This one really emphasizes how we can manage side-effects each in its contained callback, instead of having complex `didMount`/`didUpdate` methods.

### CODE REVIEW

File: src/panel/category/CategoryPanel.jsx [hunk 1]
```diff
@@ -16,165 +16,137 @@ import { fire, listen, unListen } from 'src/libs/customEvents';
...
-      window.map.mb.flyTo({ zoom: 12 });
-    } else if (currentZoom > 18) { // Zoom > 18: dezoom to zoom 18
-      window.map.mb.flyTo({ zoom: 18 });
-    } else {
-      // setting the same view still triggers the moveend event
-      window.map.mb.jumpTo({ zoom: currentZoom, center: window.map.mb.getCenter() });
-    }
-  }
+  useEffect(() => {
+    Telemetry.add(Telemetry.POI_CATEGORY_OPEN, { category: poiFilters.category });
```
[the remark this answers is not in the document]
Comment (bbecquet) [reply src/panel/category/CategoryPanel.jsx:105]: Right. I'll send it only in the cases where `category` is defined.

File: src/panel/category/CategoryPanel.jsx [hunk 2]
```diff
@@ -16,165 +16,136 @@ import { fire, listen, unListen } from 'src/libs/customEvents';
...
+    }, DEBOUNCE_WAIT, { leading: true });
 
-  sendTelemetry(prevProps) {
-    const { category } = this.props.poiFilters;
-    if (category && category !== prevProps?.poiFilters?.category) {
-      Telemetry.add(Telemetry.POI_CATEGORY_OPEN, { category });
-    }
-  }
+    const mapMoveHandler = listen('map_moveend', fetchData);
+    return () => { unListen(mapMoveHandler); };
```
Comment (G-Ray) [inline src/panel/category/CategoryPanel.jsx:93]: :+1:

File: src/panel/category/CategoryPanel.jsx [hunk 3]
```diff
@@ -16,165 +16,136 @@ import { fire, listen, unListen } from 'src/libs/customEvents';
...
 
 const categoryConfig = nconf.get().category;
 const MAX_PLACES = Number(categoryConfig.maxPlaces);
 const DEBOUNCE_WAIT = 100;
 
-export default class CategoryPanel extends React.Component {
-  static propTypes = {
-    poiFilters: PropTypes.object,
-    bbox: PropTypes.string,
+function fitMap(bbox) {
```
Comment (G-Ray) [inline src/panel/category/CategoryPanel.jsx:26]: :+1:

File: src/panel/category/CategoryPanel.jsx [hunk 4]
```diff
@@ -16,165 +16,136 @@ import { fire, listen, unListen } from 'src/libs/customEvents';
...
-  selectPoi = poi => {
-    const { poiFilters } = this.props;
-    const { pois } = this.state;
+  const selectPoi = poi => {
     fire('click_category_poi', { poi, poiFilters, pois });
-  }
+  };
 
-  highlightPoiMarker = (poi, highlight) => {
+  const highlightPoiMarker = (poi, highlight) => {
```
Comment (G-Ray) [inline src/panel/category/CategoryPanel.jsx:114]: We can optionally move these functions which do not need to be inside the rendering function to the top. I approve this PR anyway, as it seems to works well !

**Review by G-Ray — APPROVED** [review]

**Pull request merged**